clojure-mode.el

```diff
@@ -1824,6 +1836,114 @@ When BUT-LAST is passed the last expression is not threaded."
   (interactive "P")
   (clojure--thread-all "->> " but-last))
 
+;;; Cycling stuff
+
+(defcustom clojure-use-metadata-for-privacy nil
+  "If nil, `clojure-cycle-privacy' will use (defn- f []).
+If t, it will use (defn ^:private f [])."
+  :package-version '(clojure-mode . "5.5.0")
+  :safe #'booleanp
+  :type 'boolean)
+
+;;;###autoload
+(defun clojure-cycle-privacy ()
+  "Make public the current private def, or vice-versa.
+
+See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-cycle-privacy"
+  (interactive)
+  (save-excursion
+    (ignore-errors (forward-char 7))
+    (search-backward-regexp "\\((defn-? \\)\\|\\((def \\)")
+    (cond
+     ((and clojure-use-metadata-for-privacy
+           (looking-at "(defn ^:private"))
+      (forward-char 5)
+      (delete-char 10))
+     ((and (not clojure-use-metadata-for-privacy)
+           (looking-at "(defn-"))
+      (forward-char 5)
+      (delete-char 1))
+     ((and clojure-use-metadata-for-privacy
+           (looking-at "(defn"))
+      (forward-char 5)
+      (insert " ^:private"))
+     ((and (not clojure-use-metadata-for-privacy)
+           (looking-at "(defn"))
+      (forward-char 5)
+      (insert "-"))
+     ((looking-at "(def ^:private")
+      (forward-char 5)
+      (delete-char 10))
+     ((looking-at "(def ")
+      (forward-char 5)
+      (insert "^:private ")))))
+
+;;;###autoload
+(defun clojure-cycle-collection ()
+  "Convert the coll at (point) from (x) -> {x} -> [x] -> -> #{x} -> (x) recur.
+
+See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-cycle-coll"
+  (interactive)
+  (save-excursion
+    (while (and
+            (> (point) 1)
+            (not (eq (string-to-char "(") (char-after)))
```

Comment:
Actually. I'd prefer to replace this line and the 3 below with a single `looking-at`.